File: java/socketd-transport/socketd-transport-neta/src/main/java/org/noear/socketd/transport/neta/tcp/TcpAioProvider.java
package org.noear.socketd.transport.neta.tcp;

import org.noear.socketd.transport.client.Client;
import org.noear.socketd.transport.client.ClientConfig;
import org.noear.socketd.transport.client.ClientProvider;
import org.noear.socketd.transport.server.Server;
import org.noear.socketd.transport.server.ServerConfig;
import org.noear.socketd.transport.server.ServerProvider;

/**
 * @author noear
 * @since 2.3
 */
public class TcpAioProvider implements ClientProvider, ServerProvider {
    @Override
    public String[] schemas() {
        return new String[]{"tcp", "tcps", "tcp-neta", "sd:tcp", "sd:tcps", "sd:tcp-neta"};
    }

    @Override
    public Server createServer(ServerConfig serverConfig) {
        return new TcpAioServer(serverConfig);
    }

    @Override
    public Client createClient(ClientConfig clientConfig) {
        return new TcpAioClient(clientConfig);
    }
}

File: python/socketd/socketd_aio_tcp/TcpAioProvider.py
from socketd.transport.client.ClientConfig import ClientConfig
from socketd.transport.client.ClientProvider import ClientProvider
from socketd.transport.server.ServerConfig import ServerConfig
from socketd.transport.client.Client import Client
from socketd.transport.server.Server import Server
from socketd.transport.server.ServerProvider import ServerProvider
from socketd_aio_tcp import TCPAIOServer, TcpAioClient


class TcpAioProvider(ClientProvider, ServerProvider):

    def schema(self) -> list[str]:
        return ["tcp", "tcp-python", "sd:tcp"]

    def create_server(self, serverConfig: ServerConfig) -> Server:
        return TCPAIOServer(serverConfig)

    def create_client(self, clientConfig: ClientConfig) -> Client:
        return TcpAioClient(clientConfig)
